back/src/domains/core/UseCase.ts

```diff
@@ -16,6 +16,75 @@ import { UnitOfWorkPerformer } from "./unit-of-work/ports/UnitOfWorkPerformer";
 
 const logger = createLogger(__filename);
 
+type CreateTransactionalUseCase = <
+  Input,
+  Output = void,
+  UserIdentity = void,
+  Dependencies = void,
+>(
+  {
+    useCaseName,
+    inputSchema,
+  }: {
+    useCaseName: string;
+    inputSchema: z.ZodSchema<Input>;
+  },
+  cb: (
+    params: Input,
+    { uow, deps }: { uow: UnitOfWork; deps: Dependencies },
+    userIdentity: UserIdentity,
+  ) => Promise<Output>,
+) => (
+  config: {
+    uowPerformer: UnitOfWorkPerformer;
+  } & (Dependencies extends void
+    ? Record<string, any>
+    : { deps: Dependencies }),
+) => {
+  useCaseName: string;
+  execute: (params: Input, userIdentity: UserIdentity) => Promise<Output>;
+};
+
+export const createTransactionalUseCase: CreateTransactionalUseCase =
+  ({ useCaseName, inputSchema }, cb) =>
```

Comment:
```suggestion
  ({ name, inputSchema }, cb) =>
```

C'est une suggestion : répéter useCase ici me semble redondant et pas forcément utile.